### PR TITLE
Fix issues with Media Tab & performance improvement.

### PR DESCRIPTION
**Pull request by eraser2021999**
Signed-off-by: Bogdan Coticopol <bcoticopol@gmail.com>
References: 
#10882 Slow Media Tab View 
#11281

For more info, please check the discussion at the above references.
CC: @AlvaroBrey @starypatyk

### CODE REVIEW

**Comment by eraser2021999** [comment]
Are there any news regarding testing and possible moving this change to the release?

File: app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java [hunk 1]
```diff
@@ -51,7 +52,8 @@ public class GallerySearchTask extends AsyncTask<Void, Void, GallerySearchTask.R
     private final WeakReference<GalleryFragment> photoFragmentWeakReference;
     private final FileDataStorageManager storageManager;
     private final int limit;
-    private final long startDate;
+    @Deprecated
+    private final long startDate = 0; // we don't use startDate anymore, only endDate and limit
```
Comment (starypatyk) [inline app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java:56]: As the `startDate` variable is no longer used, I would suggest to remove it entirely.

File: app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java [hunk 2]
```diff
@@ -143,55 +137,54 @@ private long findLastTimestamp(ArrayList<RemoteFile> remoteFiles) {
...
-                                                                                        storageManager.getGalleryItems(startDate * 1000L,
-                                                                                                                       endDate * 1000L));
-        List<OCFile> filesToAdd = new ArrayList<>();
-        List<OCFile> filesToUpdate = new ArrayList<>();
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+        Log_OC.d(this, "parseMedia - start: " + dateFormat.format(new Date(startDate * 1000L)) + " - " + dateFormat.format(new Date(endDate * 1000L)));
+
+        List<OCFile> localFiles = storageManager.getGalleryItems(startDate * 1000L, endDate * 1000L);
+
+        for (OCFile localFile: localFiles) {
```
Comment (starypatyk) [inline app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java:145]: This logging section is useful during debugging, but rather too heavy to leave it in final code. I would suggest to remove it.

File: app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java [hunk 3]
```diff
@@ -143,55 +137,54 @@ private long findLastTimestamp(ArrayList<RemoteFile> remoteFiles) {
...
+        Map<String, OCFile> localFilesMap = RefreshFolderOperation.prefillLocalFilesMap(null, localFiles);
+
+        long filesAdded = 0, filesUpdated = 0, filesDeleted = 0, unchangedFiles = 0;
 
-        OCFile localFile;
         for (Object file : remoteFiles) {
             OCFile ocFile = FileStorageUtils.fillOCFile((RemoteFile) file);
-
-            localFile = localFilesMap.remove(ocFile.getRemotePath());
+            Log_OC.d(this, "remote file: modified: " + dateFormat.format(new Date(ocFile.getModificationTimestamp())) + " path: " + ocFile.getRemotePath());
```
Comment (starypatyk) [inline app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java:155]: Same as above.

File: app/src/main/java/com/owncloud/android/ui/fragment/GalleryFragment.java [hunk 4]
```diff
@@ -65,10 +65,12 @@ public class GalleryFragment extends OCFileListFragment implements GalleryFragme
 
     private boolean photoSearchQueryRunning = false;
     private AsyncTask<Void, Void, GallerySearchTask.Result> photoSearchTask;
-    private long startDate;
+    @Deprecated
+    private long startDate = 0; // we don't use the startDate anymore, we use only endDate and row limit
```
Comment (starypatyk) [inline app/src/main/java/com/owncloud/android/ui/fragment/GalleryFragment.java:69]: Same as in `GallerySearchTask` - as this is unused I would suggest to remove the variable.

File: app/src/main/java/com/owncloud/android/ui/fragment/GalleryFragment.java [hunk 5]
```diff
@@ -65,10 +65,12 @@ public class GalleryFragment extends OCFileListFragment implements GalleryFragme
...
     private long endDate;
-    private long daySpan = 30;
-    private int limit = 300;
+    @Deprecated
+    private long daySpan = 30; //not used anymore
```
Comment (starypatyk) [inline app/src/main/java/com/owncloud/android/ui/fragment/GalleryFragment.java:72]: Same as above.

**Comment by starypatyk** [comment]
@eraser2021999

I have run some tests and with these changes the Media tab is indeed much more stable. :+1: I have not noticed negative side-effects.

I have put a few small comments above regarding unused variables - should be trivial to change. Also a trivial change suggested by Codacy Static Code Analysis: https://github.com/nextcloud/android/pull/11321/files#annotation_8372803674

I have managed to crash the app in the emulator - but this seems to be unrelated to this PR. On the actual phone the app was stable - although this was on an old version of Android (7).  The previous version was even easier to crash. This will need to be investigated separately.

Generally - looks good to me.

PS. @eraser2021999 - When making updates, please **do not** sync your master branch with master in nextcloud/android. Hopefully this will let us avoid problems with Git.

File: app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java (hunk 2 above)
Comment (AlvaroBrey) [reply app/src/main/java/com/owncloud/android/ui/asynctasks/GallerySearchTask.java:145]: (or maybe fence it with a `if (BuildConfig.DEBUG)`

**Comment by eraser2021999** [comment]
@starypatyk @AlvaroBrey 
I refactored the code based on your remarks. I retested the application with both scenarios (empty gallery and with some old media already loaded in the local storage via folder navigation), everything seems to work as expected.

I kindly ask you to retest, to make sure I didn't introduced new bugs and, if everything is ok.

Thanks a lot!

**Review by starypatyk — APPROVED** [review]
Tested briefly, looks fine. :+1:

Thanks @eraser2021999!

**Comment by eraser2021999** [comment]
Any update on the release progress?

**Review by AlvaroBrey — APPROVED** [review]

**Comment by AlvaroBrey** [comment]
@tobiasKaminsky can you review?

**Comment by eraser2021999** [comment]
Wow, really 2 months for a code review...do you plan to implement it? @szaimen @AlvaroBrey

**Review by tobiasKaminsky — APPROVED** [review]

**Comment by tobiasKaminsky** [comment]
Sorry for the delay, we are currently a bit understaffed… :S

**Comment by szaimen** [comment]
Thanks a lot for this PR! Just tested it and works much faster! 🚀🎉

**Comment by hanisntsolo** [comment]
Anything for iOS ?? There it's still an issue.

**Pull request merged**